### PR TITLE
ADR - Managing secrets in Managed workspaces

### PR DESCRIPTION
**Pull request by scoheb**

### CODE REVIEW

File: ADR/0031-managing-secrets-in-managed-workspaces.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Accepted
+
+## Context
+Currently, secrets are managed in managed workspaces via SPI's RemoteSecrets. This involves creating RemoteSecret resources in addition to creating and uploading Secrets to be linked to the RemoteSecrets. This secret data does not exist in any gitops repos nor within any secret service provider. It only exists locally on one's laptop. This poses a problem when a service or team needs to rotate a secret or even verify if the secret is valid.
+
+## Decision
+Managed workspaces will use ExternalSecrets via the External Secrets Operator. This will permit the tight integration with Vault that is already used in other systems.
+
+## Consequence
+- Users will need to created and administer workspace secrets using a combination of Vault and ExternalSecret resources.
```
Comment (arewm) [inline ADR/0031-managing-secrets-in-managed-workspaces.md:14]: nit:
```suggestion
- Users will need to create and administer workspace secrets using a combination of Vault and ExternalSecret resources.
```

**Review by arewm — APPROVED** [review]

File: ADR/0031-managing-secrets-in-managed-workspaces.md (hunk 1 above)
Comment (arewm) [reply ADR/0031-managing-secrets-in-managed-workspaces.md]: @scoheb , while this is outdated according to GitHub, the proposal is still valid.

Comment (scoheb) [reply ADR/0031-managing-secrets-in-managed-workspaces.md]: sorry. a rebase removed your suggestion. updated

Comment (arewm) [reply ADR/0031-managing-secrets-in-managed-workspaces.md]: All good, thanks for fixing it!

File: ADR/0032-managing-secrets-in-managed-workspaces.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# 32. Managing Secrets in Managed Workspaces
+Date: 2023-11-22
+
+## Status
+Accepted
+
+## Context
+Currently, secrets are managed in managed workspaces via SPI's RemoteSecrets. This involves creating RemoteSecret resources in addition to creating and uploading Secrets to be linked to the RemoteSecrets. This secret data does not exist in any gitops repos nor within any secret service provider. It only exists locally on one's laptop. This poses a problem when a service or team needs to rotate a secret or even verify if the secret is valid.
```
Comment (gazarenkov) [inline ADR/0032-managing-secrets-in-managed-workspaces.md:8]: > This secret data does not exist in any gitops repos nor within any secret service provider.

Is it considered that this data will be pre-populated (apparently by dedicated person, like system admin) or Managed workspaces consider some self-servicing in this regard? 
If latter it become more interesting b/c ESO is likely designed for the first case.

**Review by gbenhaim — APPROVED** [review]

**Review by ralphbean — APPROVED** [review]

File: ADR/0032-managing-secrets-in-managed-workspaces.md (hunk 2 above)
Comment (arewm) [reply ADR/0032-managing-secrets-in-managed-workspaces.md:8]: @gazarenkov , do you need this question answered still?

**Pull request closed without merging**